integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py

```diff
@@ -383,6 +384,77 @@ def query_by_embedding(
                 document.score = score
         return results
 
+    def _query_hybrid(
+        self,
+        query_embedding: List[float],
+        query_sparse_embedding: SparseEmbedding,
+        filters: Optional[Dict[str, Any]] = None,
+        top_k: int = 10,
+        return_embedding: bool = False,
+    ) -> List[Document]:
+        """
+        Retrieves documents based on both dense and sparse embeddings
+         and fuses the results using Reciprocal Rank Fusion.
+
+        This method is not mean to be part of the public interface of
+        `QdrantDocumentStore` nor called directly.
+        `QdrantHybridRetriever` uses this method directly and is the public interface for it.
+
+        :param query_embedding: Dense embedding of the query.
+        :param query_sparse_embedding: Sparse embedding of the query.
+        :param filters: Filters applied to the retrieved Documents.
+        :param top_k: Maximum number of Documents to return.
+        :param return_embedding: Whether to return the embeddings of the retrieved documents.
+
+        :returns: List of Document that are most similar to `query_embedding` and `query_sparse_embedding`.
+        """
+
+        # this implementation is based on that of Python Qdrant client
+        # https://github.com/qdrant/qdrant-client/blob/8e3ea58f781e4110d11c0a6985b5e6bb66b85d33/qdrant_client/qdrant_fastembed.py#L519
+        if not self.use_sparse_embeddings:
+            message = (
+                "You are trying to query using sparse embeddings, but the Document Store "
+                "was initialized with `use_sparse_embeddings=False`. "
+            )
+            raise QdrantStoreError(message)
+
+        qdrant_filters = convert_filters_to_qdrant(filters)
+
+        sparse_request = rest.SearchRequest(
```

Comment:
Should we catch exceptions here, and `raise from`? Not sure if a finer grain is helpful so feel free to ignore

Reply:
Now I tried doing this for the cases that make the most sense to me